core/snippets/snippets_parser.py

```diff
@@ -192,17 +192,13 @@ def parse_document(
     file: str, line: int, optional_body: bool, text: str
 ) -> Union[SnippetDocument, None]:
     parts = re.split(r"^-\r?\n?$", text, flags=re.MULTILINE)
```

Comment:
Can we more simply use `text.split("-\n", 1)` here? `\r\n` should be translated to `\n` by `open` universal newlines mode.

Reply:
I also want to split if there is no trailing newline. That would be a snippet with an empty body. An unlikely case, but I still want to support it.